lib/spack/spack/subprocess_context.py

```diff
@@ -94,17 +94,27 @@ def __init__(self):
             self.platform = spack.architecture.platform
             self.test_patches = store_patches()
 
-            # TODO: transfer spack.store.store? note that you should not
-            # transfer spack.store.store and spack.store.db: 'db' is a
-            # shortcut that accesses the store (so transferring both can
-            # create an inconsistency). Some tests set 'db' directly, and
-            # others set 'store'
+            self.store_root = spack.store.store.root
+            self.store_unpadded_root = spack.store.store.unpadded_root
+            self.store_projections = spack.store.store.projections
+            self.store_hash_length = spack.store.store.hash_length
```

Comment:
This appeared to set `store.db` and `store` without addressing the TODO above, so I'm trying to figure out why tests like `test_installed_upstream` are still passing (they monkeypatch `store.db`, which is undone by this) - my current guess is that the prefixes are determined at the time of the installation and that particular test is only checking prefixes rather than doing DB queries. Overall though those tests are made somewhat fragile by this update and I think this PR will need to address that fragility somehow (e.g. by removing the monkeypatching calls in `test/install.py`).

Reply:
Calls to `monkey patch.setattr` have been removed in a644293